src/inmanta/ast/statements/define.py

```diff
@@ -576,7 +576,10 @@ def evaluate(self) -> None:
         else:
             left_end = None
 
-        if self.right[1] is not None:
+        if right == left and str(self.left[1]) == str(self.right[1]) and self.right[1] is not None:
```

Comment:
This condition is strictly stronger than the one on line 582. Wouldn't it be cleaner to write it as follows? This might only be my personal preference, I'm not sure if there is a convention for this.
```
if self.right[1] is not None:
    if right == left and str(self.left[1]) == str(self.right[1]):
        # relation is its own inverse
    else:
        # plain relation
```